modules: lvgl: allow usage of zephyr memory-region for mem / vdb

avolmat-st opened the pull request:
Allow selection of a custom section for memory-poll and / or VDB based on zephyr memory-regions.
This takes advantages that zephyr,memory-region automatically have sections being created, hence allowing to easily indicate into which memory-region to store data.

Commit:
modules: lvgl: allow usage of zephyr memory-region for mem / vdb

Allow selection of a custom section for memory-poll and / or VDB
based on zephyr memory-regions.
This takes advantages that zephyr,memory-region automatically have
sections being created, hence allowing to easily indicate into which
memory-region to store data.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

faxe1008 commented on modules/lvgl/Kconfig.memory:
```diff
@@ -52,6 +52,22 @@ config LV_Z_MEMORY_POOL_CUSTOM_SECTION
+	  Place LVGL memory pool in custom generic section automatically
+	  created by Zephyr when compatible zephyr,memory-region is used.
+	  LV_Z_MEMORY_POOL_CUSTOM_SECTION must be disabled in order to use
+	  generic section.
```
If they are exclusive, rather make them a choice symbol.

avolmat-st replied:
Indeed, I agree. I just tried that but this will make have a much bigger impact since the CONFIG_LV_Z_MEMORY_POOL_CUSTOM is set as part of fomr Kconfig.default within the zephyr tree and moving this into a choice section do not allow this to be set via a Kconfig.default.

ex:
st_b_lcd40_dsi1_mb1166/boards/stm32h757i_eval_stm32h757xx_m7.defconfig:16, modules/lvgl/Kconfig.memory:105, modules/lvgl/Kconfig.memory:105) will have no effect, as defaults do not affect choice symbols

faxe1008 commented on modules/lvgl/Kconfig.memory:
```diff
@@ -52,6 +52,22 @@ config LV_Z_MEMORY_POOL_CUSTOM_SECTION
+	  LV_Z_MEMORY_POOL_CUSTOM_SECTION must be disabled in order to use
+	  generic section.
+
+config LV_Z_MEMORY_POOL_CUSTOM_GENERIC_SECTION_NAME
```
Dont know about the naming of the symbol, "custom generic" seems to contradict itsself a bit imo. Maybe something involving the memory region concept?

faxe1008 commented on modules/lvgl/Kconfig.memory:
```diff
@@ -92,6 +108,22 @@ config LV_Z_VDB_CUSTOM_SECTION
 	  rendering buffers to a custom location, such as tightly coupled or
 	  external memory.
 
+config LV_Z_VDB_CUSTOM_GENERIC_SECTION
```
Same here, choice symbol and maybe another name.

avolmat-st commented:
Partially updated without addition of choice for the time being. I will see for the choice / endchoice section, which will probably will make the PR bigger in order to also update existing Kconfig.default / conf files.

avolmat-st commented:
This idea of this PR would be to avoid having to prepare linker scripts such as 

boards/st/stm32h747i_disco/dc_ram.ld, boards/st/stm32h757i_eval/dc_ram.ld, boards/st/stm32h7b3i_dk/dc_ram.ld

for all boards, which also do not give flexibility on which section to use.

avolmat-st commented:
Update following @JarmouniA proposal for config name and added depend !... to avoid possibility to have both mode enabled at the same time. I only put it on one side (ZEPHYR_REGION depends on !CUSTOM_SECTION) to avoid dependency loop.

Commit:
boards: st: remove .lvgl_buf related linker scripts on STM32

Following introduction of LV_Z_MEMORY_POOL_ZEPHYR_REGION and
LV_Z_VDB_ZEPHYR_REGION, remove the usage of the option
LV_Z_VDB_CUSTOM_SECTION and the related linker scripts for
STM32 boards & shields.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

uLipe reviewed (commented):
Besides the others comments, LGTM.

erwango commented:
@avolmat-st Please rebase to fix CI issue.

avolmat-st commented:
Thanks @erwango, now rebased.

kartben reviewed (commented):
please mention new Kconfig options in release notes
